src/screens/Inventory/CreateInventoryUpdate.tsx

```diff
@@ -0,0 +1,77 @@
+import { createStyles, Theme, withStyles } from '@material-ui/core/styles';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { Redirect, RouteComponentProps, useHistory } from 'react-router-dom';
+import BaseScreen from '../../components/BaseComponents/BaseScreen';
+import BaseScrollView from '../../components/BaseComponents/BaseScrollView';
+import Button from '../../components/Button';
+import TextField from '../../components/TextField';
+import { createInventoryUpdateAndUpdateInventory } from '../../lib/airtable/request';
+import { selectCurrentInventory, selectCurrentInventoryProduct } from '../../lib/redux/inventoryData';
+import { selectCurrentUserId } from '../../lib/redux/userData';
+import { getInventoryLastUpdated } from '../../lib/utils/inventoryUtils';
+import InventoryInfo from './components/InventoryInfo';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    content: {
+      color: theme.palette.text.primary,
+      display: 'flex',
+      flexDirection: 'column',
+    },
+  });
+
+interface CreateInventoryUpdateProps extends RouteComponentProps {
+  classes: { content: string };
+}
+
+function CreateInventoryUpdate(props: CreateInventoryUpdateProps) {
+  const { classes } = props;
+  const history = useHistory();
+  const userId = useSelector(selectCurrentUserId);
+  const inventory = useSelector(selectCurrentInventory);
+  const product = useSelector(selectCurrentInventoryProduct);
+
+  const [updatedAmount, setUpdatedAmount] = useState(0.0);
+
+  // Redirect to InventoryMain if undefined
+  if (!userId || !inventory || !product) {
+    return <Redirect to={'/inventory'} />;
+  }
+
+  // TODO @wangannie: add better edge case handling
+  const handleUpdatedAmount = (event: React.ChangeEvent<{ value: unknown }>) => {
+    setUpdatedAmount(parseFloat(event.target.value as string) || 0);
+  };
+
+  const handleSubmit = async (event: React.MouseEvent) => {
```

Comment:
This also has the bug where if you click multiple times in quickly, unintended behavior occurs. Did you want to fix that behavior all at once sometime later?

Reply:
Fixed with a loading circle + disabling the button! It's a temporary fix and we'll probably need to check with Ashley on the design, but it prevents double clicks
![image](https://user-images.githubusercontent.com/21160510/113441800-c50f5200-93a3-11eb-899d-2f3432c89f8c.png)